var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -89,9 +89,11 @@ class Magma(CMakePackage, CudaPackage, ROCmPackage):
     # https://github.com/icl-utk-edu/magma/issues/7
     conflicts("^cuda@12.6:", when="@:2.8.0")
 
-    # Many cuda_arch values are not yet recognized by MAGMA's CMakeLists.txt
-    for target in [10, 11, 12, 13, 21, 32, 52, 53, 61, 62, 72, 86]:
-        conflicts(f"cuda_arch={target}")
+    # Many cuda_arch values were not yet recognized by MAGMA's CMakeLists.txt
+    with when("@:2.9.0"):
+        # All cuda_arch values are supported in latest 2.9.0 release
+        for target in [10, 11, 12, 13, 21, 32, 52, 53, 61, 62, 72, 86]:
+            conflicts(f"cuda_arch={target}")
 
     # Some cuda_arch values had support added recently
     conflicts("cuda_arch=37", when="@:2.5", msg="magma: cuda_arch=37 needs a version > 2.5")
```

Comment:
Perhaps instead we should have an individual line for each version where a new `cuda_arch` is supported. I made the changes in the way that I did based on quite a few assumptions about how magma works, which might be wrong.

Reply:
Aren't version specifiers inclusive, so this would still include 2.9.0? That is, should it be @:2.8.0 instead?

Stylistically, I wouldn't say "_latest_ 2.9.0 release", just "2.9.0 release", so the comment isn't out-of-date when the next release comes out.

Prior to 2.9.0, MAGMA's CMakeLists.txt explicitly listed all the CUDA arch versions. Starting with 2.9.0, any sm_XY[Z] value is accepted (see https://github.com/icl-utk-edu/magma/commit/8f1994d46fff7114fc47827b8e25f5e3630436f4), so there shouldn't be a need to maintain such a detailed list of supported architectures for future MAGMA releases.